### PR TITLE
Several connector updates

### PR DESCRIPTION
**Pull request by LucGenetier**
Avoid throwing when a connector isn't supported for some reason - use IsSupported/NotSupportedReason instead
Avoid function names with curly brackets or dots (just remove those)
Properly manage httpClient when it has an absolute path (not root/empty)
Fix a NullRef when only HTTP server is provided

### CODE REVIEW

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/tests/Microsoft.PowerFx.Connectors.Tests/LivePublicSwaggerTests.cs [hunk 1]
```diff
@@ -169,6 +169,33 @@ public async Task RealTest3()
...
+            OpenApiDocument docWorldTime = await ReadSwaggerFromUrl(@"https://api.apis.guru/v2/specs/worldtimeapi.org/20210108/openapi.json").ConfigureAwait(false);
+
+            using var clientXkcd = new HttpClient() { BaseAddress = new Uri(@"http://xkcd.com/") };
+            using var clientWorldTime = new HttpClient() { BaseAddress = new Uri(@"http://worldtimeapi.org/api/") };
+            var funcsXkcd = config.AddService("Xkcd", docXkcd, clientXkcd, new ConnectorSettings() { IgnoreUnknownExtensions = true });
+            var funcsWorldTime = config.AddService("WorldTime", docWorldTime, clientWorldTime, new ConnectorSettings() { IgnoreUnknownExtensions = true });
+
+            var engine = new RecalcEngine(config);
+
+            FormulaValue fv1 = engine.Eval(@"Xkcd.comicIdinfo0json(1)");
```
Comment (MikeStall) [inline src/tests/Microsoft.PowerFx.Connectors.Tests/LivePublicSwaggerTests.cs:186]: > engine.Eval(@"Xkcd.comicIdinfo0json(1)"); [](http://example.com/codeflow?start=30&length=42)

Can we access the return result from the expression?

eg, 
`Xkcd.comicIdinfo0json(1).transcript`


`First(WorldTime.timezone()).Value`

**Review by MikeStall — APPROVED** [review]
:shipit:

**Review by MikeStall — APPROVED** [review]
:shipit:

**Comment by LucGenetier** [comment]
✅ No public API change.

**Pull request merged**